app/views/portal/show.rb

```diff
@@ -0,0 +1,871 @@
+module Portal
+  class Show < ApplicationView
+    attr_accessor :document, :debug
+
+    def page_title
+      [@document.fetch(:title, ['']).join(', '), 'Europeana'].compact.join(' - ')
+    end
+
+    def navigation
+      query_params = current_search_session.try(:query_params) || {}
+
+      if search_session['counter']
+        per_page = (search_session['per_page'] || default_per_page).to_i
+        counter = search_session['counter'].to_i
+
+        query_params[:per_page] = per_page unless search_session['per_page'].to_i == default_per_page
+        query_params[:page] = ((counter - 1) / per_page) + 1
+      end
+
+      back_link_url = if query_params.empty?
+                        search_action_path(only_path: true)
+                      else
+                        url_for(query_params)
+                      end
+
+      navigation = {
+        next_prev: {
+          prev_text: t('site.object.nav.prev'),
+          back_url: back_link_url,
+          back_text: t('site.object.nav.return-to-search'),
+          next_text: t('site.object.nav.next')
+        }
+      }
+      if @previous_document
+        navigation[:next_prev].merge!(
+          prev_url: document_path(@previous_document, format: 'html'),
+          prev_link_attrs: [
+            {
+              name: 'data-context-href',
+              value: track_document_path(@previous_document, session_tracking_path_opts(search_session['counter'].to_i - 1))
+            }
+          ]
+        )
+      end
+      if @next_document
+        navigation[:next_prev].merge!(
+          next_url: document_path(@next_document, format: 'html'),
+          next_link_attrs: [
+            {
+              name: 'data-context-href',
+              value: track_document_path(@next_document, session_tracking_path_opts(search_session['counter'].to_i + 1))
+            }
+          ]
+        )
+      end
+      navigation.merge(helpers.navigation)
+    end
+
+    def content
```

Comment:
Assignment Branch Condition size for content is too high. [102/15]<br>Method has too many lines. [346/10]